esmvalcore/preprocessor/_time.py

```diff
@@ -497,15 +497,13 @@ def regrid_time(cube, frequency):
     iris.cube.Cube
         cube with converted time axis and units.
     """
-    # fix calendars
-    cube.coord('time').units = cf_units.Unit(
-        cube.coord('time').units.origin,
-        calendar='gregorian',
-    )
-
     # standardize time points
     time_c = [cell.point for cell in cube.coord('time').cells()]
-    if frequency == 'mon':
+    if frequency == 'yr':
+        cube.coord('time').cells = [
```

Comment:
```suggestion
        cells = [
```
Don't replace the [cube.coord('time').cells](https://scitools.org.uk/iris/docs/v2.2.1/iris/iris/coords.html#iris.coords.Coord.cells) function with an array. Just use `cells` or some other name for your array.